gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/service/TaskService.java

```diff
@@ -0,0 +1,238 @@
+                    }
+                    if (task.getProcessFiles().isEmpty()) {
+                        task.getProcessEvents().clear();
+                    } else {
+                        addFileEventToTask(task, FileEventType.DELETED, processFile);
+                    }
+                    return new TaskWithStatusUpdate(task, statusUpdated);
+                })
+                .collect(Collectors.toSet());
+    }
+
+    public TaskWithStatusUpdate selectFile(final OffsetDateTime timestamp,
+                                           final String filetype,
+                                           final String filename) {
+        Task task = taskRepository.findByTimestamp(timestamp).orElseThrow(TaskNotFoundException::new);
+        if (doesStatusBlockFileSelection(task.getStatus())) {
+            throw new TaskManagerException("Status of task does not allow to change selected file");
+        }
+
+        final ProcessFile processFile = task.getAvailableInputs(filetype)
+                .stream()
+                .filter(pf -> filename.equals(pf.getFilename()))
+                .findAny()
+                .orElseThrow(ProcessFileNotFoundException::new);
+        task.selectProcessFile(processFile);
+
+        String message = String.format("Manual selection of another version of %s : %s", filetype, filename);
+        OffsetDateTime now = OffsetDateTime.now(taskManagerConfigurationProperties.getProcessTimezone());
+        task.addProcessEvent(now, "INFO", message, serviceName);
+
+        boolean doesStatusNeedReset = doesStatusNeedReset(task.getStatus());
+        if (doesStatusNeedReset) {
+            task.setStatus(TaskStatus.READY);
+        }
+        return new TaskWithStatusUpdate(task, doesStatusNeedReset);
+    }
+
+    private boolean doesStatusNeedReset(final TaskStatus status) {
+        return TaskStatus.SUCCESS == status || TaskStatus.ERROR == status || TaskStatus.INTERRUPTED == status;
+    }
+
+    private boolean doesStatusBlockFileSelection(final TaskStatus status) {
+        return TaskStatus.RUNNING == status || TaskStatus.PENDING == status || TaskStatus.STOPPING == status;
+    }
+
+    ////////////////////////////
+    // RUN HISTORY MANAGEMENT //
+    ////////////////////////////
+
+    public Task addNewRunAndSaveTask(OffsetDateTime timestamp) {
+        Task task = taskRepository.findByTimestamp(timestamp).orElseThrow(TaskNotFoundException::new);
+        ProcessRun processRun = new ProcessRun(task.getProcessFiles().stream().filter(ProcessFile::isInputFile).toList());
+        task.addProcessRun(processRun);
+        return taskRepository.save(task);
+    }
+
+    static void removeUnavailableProcessFileFromTaskRunHistory(ProcessFile processFile, Task task, FileEventType fileEventType) {
+        if (FileEventType.UPDATED.equals(fileEventType) || FileEventType.DELETED.equals(fileEventType)) {
```

Comment:
Perhaps use "==" to compare Enum, as you've done L213 & L217, so it is coherent across the class (or change it the other way around)